library/rmc_restrictions/src/lib.rs

```diff
@@ -0,0 +1,36 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+
+// Data structures to represent vtable trait function pointer restrictions
+use rustc_data_structures::stable_map::FxHashMap;
+use serde::{Deserialize, Serialize};
+
+/// Trait-defined method: the trait type and the vtable index of the method.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct TraitDefinedMethod {
+    // Needs to be a string to handle both the MIR and Gotoc types
+    pub trait_name: String,
+    pub vtable_idx: usize,
+}
+
+// Call sites are identified by the trait-defined method and the function location
+// - we use a wrapper function to unambigiously point to call site locations.
+#[derive(Debug, Clone, PartialEq, Serialize, Deserialize)]
+pub struct CallSite {
+    pub trait_method: TraitDefinedMethod,
+    pub function_location: String,
```

Comment:
What is the format of this

Reply:
It's just the name now, renamed the variable to make clear